src/wasm-stack.h

```diff
@@ -793,6 +804,28 @@ template<typename SubType> void BinaryenIRWriter<SubType>::visitPop(Pop* curr) {
   emit(curr);
 }
 
+template<typename SubType>
+void BinaryenIRWriter<SubType>::visitTupleMake(TupleMake* curr) {
+  for (auto* operand : curr->operands) {
+    visit(operand);
+  }
+  if (curr->type == Type::unreachable) {
```

Comment:
One of the tuple elements could be unreachable, and this looks like the proper pattern to follow here. See other simple instructions like `visitBinary` and `visitSelect` that also handle `unreachable` like this.

Reply:
The reason instructions do that is [this](https://github.com/WebAssembly/binaryen/blob/6371cf63687c3f638b599e086ca668c04a26cbbb/src/wasm-stack.h#L424-L444); does this apply here too? (I haven't checked out everything yet; I'm asking)

Reply:
Thanks for the link @aheejin !

I think that does not apply here, as the tupleMake itself gets lowered to nothing, so that problem should not be possible.

Reply:
Hmm, good point. I suppose this should be safe to remove.